platform/ios/src/MGLMapViewDelegate.h

```diff
@@ -313,6 +313,8 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)mapViewDidStopLocatingUser:(MGLMapView *)mapView;
 
+- (MGLUserLocationStyle)mapViewStyleForUserLocationView:(MGLMapView *)mapView;
```

Comment:
MGLUserLocationStyle appears to be specific to MGLFaux3DUserLocationAnnotationView. If the developer provides a custom view in `-mapView:viewForAnnotation:`, the return value of this method is ignored. Is there a more generic way to make MGLFaux3DUserLocationAnnotationView more stylable? How about a simple delegate method that says the view will get updated based on a new location? Or more generically, a method that says any annotation view will be updated by the map view?

Reply:
You are right. This delegate method will be ignored if the users implements a custom location annotation view. We use a single annotation view to render four types of visual representations. Using a styling object was IMO is the best tradeoff to handle this. I am not sure we may want to change rendering individually each visual representation for Hydrogen.

Perhaps changing the name to `mapViewStyleForDefaultUserLocationAnnotationView:` may sound more explicit?